root/backend/server/api/routes/poll.js

```diff
@@ -0,0 +1,80 @@
+const express = require('express');
+const Poll = require('../../../db/models/poll');
+
+const router = express.Router();
+
+router.route('/')
+  .get(async (req, res, next) => {
+    try {
+      const polls = await Poll.find({}).then(data => {
+        return data.map(poll => {
+          return {
+            id: poll._id.toString(),
+            title: poll.name,
+            location: poll.location
+          }
+        });
+      });
+      res.json({ response: polls });
+      res.end();
+    } catch(err) {
+      console.log(err);
+    }
+  })
+  .post(async (req, res) => {
+    const { pollName, pollDescription, pollLocation, pollDuration, pollAvailabilities } = req.body;
+  
+    try {
+      const pollData = {
+        name: pollName,
+        description: pollDescription,
+        location: pollLocation,
+        duration: parseInt(pollDuration) * 60,
+        availabilities: pollAvailabilities
+      }
+  
+      const poll = new Poll(pollData);
+      await poll.save();
+      res.end();
+    } catch(err) {
+      console.log(err);
+    }
+  })
+
+router.patch('/:pollId', async (req, res) => {
+  const { pollName, pollDescription, pollLocation, pollDuration, pollAvailabilities } = req.body;
+
+  try {
+    const pollData = {
+      name: pollName,
+      description: pollDescription,
+      location: pollLocation,
+      duration: parseInt(pollDuration) * 60,
+      availabilities: pollAvailabilities
+    }
+
+    await Poll.findOneAndUpdate({ _id: req.params.pollId }, pollData);
+    res.end();
```

Comment:
should probably be sending a success response status here. something like:

```javascript
res.status(200).json({ ...some data here if needed... });
```

or if you don't need to send back any data:

```javascript
res.sendStatus(200);
```